source/extensions/filters/http/dynamo/dynamo_stats.cc

```diff
@@ -0,0 +1,117 @@
+#include "extensions/filters/http/dynamo/dynamo_stats.h"
+
+#include <iostream>
+#include <memory>
+#include <string>
+
+#include "envoy/stats/scope.h"
+
+#include "common/stats/symbol_table_impl.h"
+
+#include "extensions/filters/http/dynamo/dynamo_request_parser.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Dynamo {
+
+DynamoStats::DynamoStats(Stats::Scope& scope, const std::string& prefix)
+    : scope_(scope), pool_(scope.symbolTable()), prefix_(pool_.add(prefix + "dynamodb")),
+      batch_failure_unprocessed_keys_(pool_.add("BatchFailureUnprocessedKeys")),
+      capacity_(pool_.add("capacity")), empty_response_body_(pool_.add("empty_response_body")),
+      error_(pool_.add("error")), invalid_req_body_(pool_.add("invalid_req_body")),
+      invalid_resp_body_(pool_.add("invalid_resp_body")),
+      multiple_tables_(pool_.add("multiple_tables")), no_table_(pool_.add("no_table")),
+      operation_missing_(pool_.add("operation_missing")), table_(pool_.add("table")),
+      table_missing_(pool_.add("table_missing")), upstream_rq_time_(pool_.add("upstream_rq_time")),
+      upstream_rq_total_(pool_.add("upstream_rq_total")) {
+  upstream_rq_total_groups_[0] = pool_.add("upstream_rq_total_unknown");
+  upstream_rq_time_groups_[0] = pool_.add("upstream_rq_time_unknown");
+  for (int i = 1; i < 6; ++i) {
+    upstream_rq_total_groups_[i] = pool_.add(fmt::format("upstream_rq_total_{}xx", i));
+    upstream_rq_time_groups_[i] = pool_.add(fmt::format("upstream_rq_time_{}xx", i));
+  }
+  RequestParser::forEachStatString([this](const std::string& str) {
+    // Thread annotation does not realize this function is only called from the
+    // constructor, so we need to lock the mutex. It's easier to just do that
+    // and it's no real penalty.
+    absl::MutexLock lock(&mutex_);
+    builtin_stat_names_[str] = pool_.add(str);
+  });
+  builtin_stat_names_[""] = Stats::StatName();
```

Comment:
It probably would be useful to remember in the builtins a few common request-code stats here too, like 200, 404, 503 or something, as we do in source/common/http/codes.cc.

However I don't know what the right set of likely status codes is for dynamo DB.

Reply:
Just add a TODO for now? In general can you add TODO/comments about where you think there might be potential perf issues? IMO it's OK in this code to not have it be optimal for now as long as we have comments on the cases in which we might have issues.